libraries/logging/iot_logging_task_dynamic_buffers.c

```diff
@@ -175,18 +175,39 @@ void vLoggingPrintf( const char * pcFormat,
                         pcTaskName = pcNoTask;
                     }
 
-                    xLength = snprintf( pcPrintString, configLOGGING_MAX_MESSAGE_LENGTH, "%lu %lu [%s] ",
-                                        ( unsigned long ) xMessageNumber++,
-                                        ( unsigned long ) xTaskGetTickCount(),
-                                        pcTaskName );
-                }
-            #else /* if ( configLOGGING_INCLUDE_TIME_AND_TASK_NAME == 1 ) */
-                {
-                    xLength = 0;
+                    xLength += snprintf( pcPrintString, configLOGGING_MAX_MESSAGE_LENGTH, "%lu %lu [%s] ",
+                                         ( unsigned long ) xMessageNumber++,
+                                         ( unsigned long ) xTaskGetTickCount(),
+                                         pcTaskName );
                 }
             #endif /* if ( configLOGGING_INCLUDE_TIME_AND_TASK_NAME == 1 ) */
         }
 
+        /* Choose the string for the log level metadata for the log message. */
+        switch( usLoggingLevel )
+        {
+            case LOG_ERROR:
+                pcLevelString = "ERROR";
+                break;
+
+            case LOG_WARN:
+                pcLevelString = "WARN";
+                break;
+
+            case LOG_INFO:
+                pcLevelString = "INFO";
+                break;
+
+            case LOG_DEBUG:
+                pcLevelString = "DEBUG";
+        }
+
+        /* Add the chosen log level information as prefix for the message. */
+        if( pcLevelString != NULL )
+        {
+            xLength += snprintf( pcPrintString + xLength, configLOGGING_MAX_MESSAGE_LENGTH - xLength, "[%s] ", pcLevelString );
```

Comment:
Should we check `snprintf` failure?

Reply:
Added an assert as negative value is returned by [snprintf](http://www.cplusplus.com/reference/cstdio/snprintf/) only for encoding errors.